src/pages/Royale/components/Arena/queries/useRoyaleArenaContractQuery.ts

```diff
@@ -34,6 +46,97 @@ const useRoyaleArenaContractQuery = (season: number, address: string, options?:
 };
 
 const getFromContract = async (
+    RoyaleContract: any,
+    season: number,
+    tokenId: string | null
+): Promise<RoyaleArenaData> => {
+    const roundInASeason = await RoyaleContract.roundInASeason(season);
+    if (tokenId !== '') {
+        const [
+            isPlayerAlive,
+            roundInASeasonStartTime,
+            roundInASeasonEndTime,
+            roundChoosingLength,
+            canCloseRound,
+            seasonFinished,
+            rewardCollectedPerSeason,
+            rounds,
+            token,
+            targetPrice,
+            position,
+            round,
+        ] = await Promise.all([
+            RoyaleContract.isTokenAliveInASpecificSeason(tokenId, season),
+            RoyaleContract.roundInASeasonStartTime(season),
+            RoyaleContract.roundInSeasonEndTime(season),
+            RoyaleContract.roundChoosingLength(),
+            RoyaleContract.canCloseRound(),
+            RoyaleContract.seasonFinished(season),
+            RoyaleContract.tokenRewardCollectedPerSeason(tokenId),
+            RoyaleContract.rounds(),
+            RoyaleContract.oracleKeyPerSeason(season),
+            RoyaleContract.targetPricePerRoundPerSeason(season, roundInASeason),
+            RoyaleContract.tokenPositionInARoundPerSeason(tokenId, roundInASeason),
+            RoyaleContract.roundInASeason(season),
+        ]);
+
+        return {
+            roundChoosingLength: Number(roundChoosingLength),
+            roundInASeasonStartTime: new Date(Number(roundInASeasonStartTime) * 1000),
+            roundInASeasonEndTime: new Date(Number(roundInASeasonEndTime) * 1000),
+            roundInASeason: Number(roundInASeason),
+            canCloseRound,
+            isPlayerAlive,
+            seasonFinished,
+            rewardCollectedPerSeason,
+            rounds: Number(rounds),
+            token: parseBytes32String(token),
+            targetPrice: ethers.utils.formatEther(targetPrice),
+            position: Number(position),
+            round,
+        };
+    } else {
+        const [
+            roundInASeasonStartTime,
+            roundInASeasonEndTime,
+            roundChoosingLength,
+            canCloseRound,
+            seasonFinished,
+            rounds,
+            token,
+            targetPrice,
+            round,
+        ] = await Promise.all([
+            RoyaleContract.roundInASeasonStartTime(season),
+            RoyaleContract.roundInSeasonEndTime(season),
+            RoyaleContract.roundChoosingLength(),
+            RoyaleContract.canCloseRound(),
+            RoyaleContract.seasonFinished(season),
+            RoyaleContract.rounds(),
+            RoyaleContract.oracleKeyPerSeason(season),
+            RoyaleContract.targetPricePerRoundPerSeason(season, roundInASeason),
+            0,
```

Comment:
was ist das?

Reply:
When tokenId === '', Promise fails therefore leading to problem with rendering royale rounds for those who don't participate. Just a quick & dirty fix before releasing.